src/query/interpret/eval.hpp

```diff
@@ -312,13 +310,13 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
       if (!frame_change_collector_->IsKeyValueCached(*cached_id)) {
         // Check only first time if everything is okay, later when we use
         // cache there is no need to check again as we did check first time
-        get_list_literal();
+        _list = get_list_literal();
         auto preoperational_checks = do_list_literal_checks();
         if (preoperational_checks) {
           return std::move(*preoperational_checks);
         }
         auto &cached_value = frame_change_collector_->GetCachedValue(*cached_id);
-        cached_value.CacheValue(*_list_ptr);
+        cached_value.CacheValue(_list);
```

Comment:
Could the list be moved?

Reply:
Why would we want to move _list here?

Reply:
There is no reason to have the `_list`, this is now just a simple temporary object. Assuming you did `auto list = get_list_literal()` above, you could then move it into the CacheValue call, so we don't do another copy. 

If I am not missing something, the list is not needed after the CacheValue call, so could be moved.

Reply:
The CacheValue is currently taking only const ref, but could be expanded to take in &&